src/Config/Method.php

```diff
@@ -0,0 +1,87 @@
+<?php
+namespace GravApi\Config;
+
+use GravApi\Helpers\ArrayHelper;
+
+/**
+ * Class Method
+ * @package GravApi\Config
+ */
+class Method
+{
+    /**
+     * @var boolean
+     */
+    protected $enabled = false;
+
+    /**
+     * @var boolean
+     */
+    protected $useAuth = true;
+
+    /**
+     * @var array
+     */
+    protected $fields = array();
+
+    /**
+     * @var array
+     */
+    protected $ignore_files = array();
+
+    public function __construct($config = null)
+    {
+        if (!isset($config) || !is_array($config)) {
+            $config = array();
+        }
+
+        $enabled = isset($config['enabled'])
+            ? $config['enabled']
+            : null;
+
+        if (is_bool($enabled)) {
+            $this->enabled = $enabled;
+        }
+
+        $auth = isset($config['auth'])
```

Comment:
Perhaps for simplicity of matching properties, this config field could be renamed to `useAuth`?